### PR TITLE
[kwokctl] Add --dry-run

### PR DESCRIPTION
**Pull request by wzshiming**
#### What type of PR is this?

/kind feature

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
[kwokctl] Add --dry-run
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: pkg/utils/dryrun/dryrun.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	"sigs.k8s.io/kwok/pkg/utils/slices"
+)
+
+// DryRun is a flag to indicate whether the program is running in dry-run mode.
+var DryRun bool
+
+// IsDryRun returns true if the program is running in dry-run mode.
+func IsDryRun() bool {
+	return DryRun
+}
```
Comment (lianghao208) [inline pkg/utils/dryrun/dryrun.go:38]: Should we move these to pkg/config/vars.go? Considering dryrun is also a config var of kwokctl like others.

Comment (wzshiming) [reply pkg/utils/dryrun/dryrun.go:38]: emmm, It should not be suitable, `pkg/config/vars.go` is for `kwok`/`kwokctl`, `dryrun` is only for `kwokctl` and contains some additional output functions that do not correspond to the config.

Comment (lianghao208) [reply pkg/utils/dryrun/dryrun.go:37]: I mean only move the DryRun variable definition and getter function to the config package.  Putting these additional output functions in an independent util package looks good to me. IMO, it's a bit odd to set and get a config value from a util package.

Comment (wzshiming) [reply pkg/utils/dryrun/dryrun.go:37]: > it's a bit odd to set and get a config value from a util package.
That makes sense, then I think we can move dryrun,file,exec to `pkg/kwokctl`.

**Pull request merged**